helm/chart/router/values.yaml

```diff
@@ -10,11 +10,14 @@ router:
     server:
       listen: 0.0.0.0:80
   args:
-  - --hot-reload
+    - --hot-reload
 
 managedFederation:
   # -- If using managed federation, the graph API key to identify router to Studio
-  apiKey: ""
+  apiKey:
+  # -- If using managed federation, use existing Secret which stores the graph API key instead of creating a new one.
+  # If set, this parameter takes precedence over `managedFederation.apiKey`
+  apiKeyExistingSecret:
```

Comment:
I feel a bit uncomfortable with this interface. I'd prefer something which makes it clear that a secret should be created (from a supplied value) or that an existing secret be used. I had a think and what I really want is:

```
managedFederation:
  secret:
  apiKey:
```
where both fields are optional.
If you specify none of them, then you aren't using managedFederation.
If you specify just the secret, then it is used as the source of apiKey.
If you specify just the apiKey, then a secret with the default name is created containing the apiKey.
If you specify both: the apiKey is written into a created secret with the name you supply. If the secret exists, that's an error (unless we can overwrite it somehow)

My helm skills are pretty basic, so I don't even know if that's feasible, but it feels like it would be a neater interface. What do you think?

Reply:
Hmm, I feel like we should use a more self-explanatory name than `secret`, maybe we could reach a compromise with `existingSecret`, what you think?

---

I think it's feasible, yeah. Just to make clear, when both are specified, you would like `apiKey` to took precedence over the `secret`? I usually see the opposed, but I don't think neither way is right or wrong

Reply:
`secretName` fails to clarify the use of an already provisioned secret and make even more confusing, IMO. I kinda get the confusion with `existingSecret` and a agree that's redundant, but that's how usually chart's values are structured, that's why documentation plays a big role in a helm chart.

It's possible to throw an error but that would mean having to create validations, which I feel are a tad overkill just to cover this case.

Reply:
Ok. Let's reset... Your original proposal seemed a bit confusing to me, but your proposition is that this is usually how chart values are structured: i.e. precedence with documentation rather than explicit choices (also possibly with documentation). I can see why, because declaration vs runtime enforcement is important in helm.

Let's rewind to your original proposal (with the fix for not creating a key if using an existing key) and say that is the solution. There must be a better name than: `apiKeyExistingSecret` though... How about just `existingSecret` or `secretWithKey` or `secretWithApiKey` or ... ?

Reply:
Great! I would prefer `existingSecret`, if that's fine with you.